### PR TITLE
Add HTTP Basic Auth support

### PR DESCRIPTION
**Pull request by timkendall**
#30

### CODE REVIEW

**Comment by leo** [comment]
Looks great! But please don't increment the version tag in `package.json` for us. Depending on what will follow, it might also be a major release... 😊

**Comment by timkendall** [comment]
Gotcha. Was wondering about that. I'll revert it when I get back to my computer later today. Great little library thanks for sharing!

**Comment by aesopwolf** [comment]
I'd be interested in seeing a slightly different API. One that pulls user/pass from environment variables automatically (perhaps `process.env.LIST_BASIC_AUTH_USER` and `process.env.LIST_BASIC_AUTH_PASS` by default). This would require only a single command line arg of `--auth`

You can then optionally supply a user/pass directly to the auth command in case you want to override looking up the environment variables. So `--auth username:password`

**Comment by timkendall** [comment]
Hmm yeah this could be good. @leo any thoughts?

**Comment by leo** [comment]
👌 

But let's only do the env vars, so that we can prevent people from hardcoding stuff. On now, env variables can be filled with secrets defined on the platform.

`LIST_USER`
`LIST_PASSWORD`

**Comment by rauchg** [comment]
Not possible with basic auth, but that would be dreamy for many many scenarios :)

File: bin/list [hunk 1]
```diff
@@ -168,6 +170,17 @@ const renderDirectory = async dir => {
 }
 
 const handler = async (req, res) => {
+  
+  if (flags.auth) {
+    const credentials = auth(req)
+
```
Comment (rauchg) [inline bin/list]: It's missing error reporting if the env vars are not set

**Comment by timkendall** [comment]
Oops. Hows that?

**Comment by leo** [comment]
Thanks!

**Comment by electerious** [comment]
How can I use HTTP Basic Auth together with `now`? The following isn't working.

```
now -e SERVE_USER=abc -e SERVE_PASSWORD=abc
```

**Comment by leo** [comment]
@electerious The flag is missing: https://github.com/zeit/serve#authentication

**Comment by electerious** [comment]
True, but how can I pass the `--auth` to the `serve` module when deploying via `now`?

**Comment by derhuerst** [comment]
@electerious It should be just like to did above.

**Comment by electerious** [comment]
But `now --auth -e SERVE_USER=abc -e SERVE_PASSWORD=abc` passes the `--auth` flag to `now`, right? It won't be forwarded to `serve`.

**Comment by leo** [comment]
Correct. You can't pass flags to serve via `now`. You need a custom package.json

**Comment by derhuerst** [comment]
@electerious The `start` script in `package.json` would usually contain the `--auth` flag. User & password would be read from the env, provided by you via `now -e …`.

**Comment by t829702** [comment]
is this feature documented how to use standalone?
https://github.com/vercel/serve#authentication

**Pull request merged**